code/jvm/src/main/kotlin/org/ionproject/codegarten/pipeline/exceptionhandling/ExceptionHandler.kt

```diff
@@ -104,6 +105,19 @@ class ExceptionHandler {
             request.requestURI,
         )
 
+    @ExceptionHandler(value = [LoopDetectedException::class])
+    private fun handleForbiddenException(
+        ex: LoopDetectedException,
+        request: HttpServletRequest
+    ): ResponseEntity<Response> =
+        handleExceptionResponse(
+            URI("/problems/loop-detected").includeHost(),
+            "Loop Detected",
+            HttpStatus.LOOP_DETECTED,

```

Comment:
Shouldn't it just be status code 500?
The [508](https://developer.cdn.mozilla.net/en-US/docs/Web/HTTP/Status/508) status code specficiation does not seem like the best option.